backend.py

```diff
@@ -25,24 +62,42 @@ def __init__(self, direction, path, coordinates):
     def __repr__(self):
         return "<Robot {} {} {}>".format(self.direction, self.path, self.coordinates)
 
-    def walk(self, distance):
+    def walk(self, distance, state):
         """
         Move a robot to new coordinates based on its direction.
         """
 
-        self.move(self.direction, distance)
+        self.move(self.direction, distance, state)
 
-    def move(self, direction, distance):
+    def move(self, direction, distance, state):
         """
         Move a robot to new coordinates according to direction of the move.
         """
-
-        (x, y) = self.coordinates
-        (new_x, new_y) = direction.coor_delta
-        x = x + (new_x * distance)
-        y = y + (new_y * distance)
-
-        self.coordinates = (x, y)
+        for __ in range(distance):
+            old_tiles = state.board[self.coordinates]
+            # On the current tile; check wall in the direction of next move.
+            for tile in old_tiles:
+                tile_move = tile.can_move_from(direction)
+                if tile_move is False:
+                    break
+            if tile_move:
+                # There is no wall, so get new coordinates
+                (x, y) = self.coordinates
+                (new_x, new_y) = direction.coor_delta
+                x = x + new_x
+                y = y + new_y
+                new_tiles = state.board[(x, y)]
+                # Check wall on the next tile in the direction of the robot movement.
+                for tile in new_tiles:
+                    tile_move = tile.can_move_to(direction)
+                    if tile_move is False:
+                        break
+                if tile_move:
+                    self.coordinates = (x, y)
+                else:
+                    break
+            else:
+                break
```

Comment:
Zde jsem se na delsi chvili ztratila: tile_move je pouzito dvakrat pro zavolani dvou ruznych metod (napada me pojmenovat jednu promennou move_from, druhou move_to). 
Radky 95 az 100 mi nedavaji smysl, prepsala jsem to zkusebne u sebe a dosahle jsem stejne funkcnosti:

```python
                # Check wall on the next tile in the direction of the robot movement.
                for tile in new_tiles:
                    move_to = tile.can_move_to(direction)
                    if not move_to:
                        break
                self.coordinates = (x, y)
```
Mozna neco prehlizim?

Reply:
- Řádeky 97, 98 tam byly navíc. Je to pozůstatek staré myšlenky - odstraněno. 
 - Řádky 99, 100: Když najdu zeď ve směru pohybu, tak ukončím prohledávání dalších políček.

Reply:
Nakonec jsem to tam ponechala a doplnila jsem vysvětlení. Jde hlavně o to, aby se zbytečně nekontrolovaly další políčka, když je nalezena zeď ve směru pohybu. Proto ti to fungovala i bez nich. Souřadnice se opravdu změní jen tehdy, když mají, ale kontrola by probíhala dál.